_clients/ruby.md

```diff
@@ -0,0 +1,528 @@
+  index: 'students'
+)
+```
+
+If you omit the request body in the `search` method, your query becomes a `match_all` query and returns all documents in the index:
+
+```ruby
+client.search(index: 'students')
+```
+
+## Boolean query
+
+The Ruby client exposes full OpenSearch query capability. In addition to simple searches that use the match query, you can create a more complex Boolean query to search for students who graduated in 2022 and sort them by last name. In the example below, search is limited to 10 documents.
+
+```ruby
+query = {
+    'query': {
+        'bool': {
+        'filter': {
+            'term': {
+                'grad_year': 2022
+                
+            }
+        }
+        }
+    },
+    'sort': {
+        'last_name': {
+            'order': 'asc'
+        }
+    }       
+}
+
+response = client.search(index: 'students', from: 0, size: 10, body: query)
+```
+
+## Multi search
+
+You can bulk several queries together and do a multi search using the `msearch` method. The following snippet searches for students whose GPAs are outside the 3.1&ndash;3.9 range:
+
+```ruby
+actions = [
+    {},
+    {query: {range: {gpa: {gt: 3.9}}}},
+    {},
+    {query: {range: {gpa: {lt: 3.1}}}}
+]
+response = client.msearch(index: 'students', body: actions)
+```
+
+## Scroll
+
+You can paginate your search results using the Scroll API:
+
+```ruby
+response = client.search(index: index_name, scroll: '2m', size: 2)
+
+while response['hits']['hits'].size.positive?
+    scroll_id = response['_scroll_id']
+    puts(response['hits']['hits'].map { |doc| [doc['_source']['first_name'] + ' ' + doc['_source']['last_name']] })
+    response = client.scroll(scroll: '1m', body: { scroll_id: scroll_id })
+end
+```
+
+At first, you issue a search query, specifying the `scroll` and `size` parameters. The `scroll` parameter tells OpenSearch how long to keep the search context. In this case, it is set to two minutes. The `size` parameter specifies how many documents you want to return in each request. 
+
+The response to the initial search query contains a `_scroll_id` that you can use to get the next set of documents. To do this, you use the `scroll` method, again specifying the `scroll` parameter and passing the `_scroll_id` in the body. You don't need to specify the query or index to the `scroll` method. The `scroll` method returns the next set of documents and the `_scroll_id`. It's important to use the latest `_scroll_id` when requesting the next batch of documents because `_scroll_id` can change between requests.
+
+## Deleting an index
+
+You can delete the index as follows:
+
+```ruby
+response = client.indices.delete(index: index_name)
+```
+
+## Sample program
+
+The following is a complete sample program that illustrates all of the concepts described above. The Ruby client's methods return responses as Ruby hashes, which are hard to read. To display JSON responses in a pretty format, the sample program uses the `MultiJson.dump` method.
```

Comment:
Replace "above" with "in the preceding sections." Replace "pretty" with "more readable".

Reply:
"Pretty" is the actual name of the format